elastic/kibana
# Add EuiCodeEditor to ES UI Shared.

This PR transplants the `EuiCodeEditor` component from EUI to the `es_ui_shared` plugin. This is a stopgap solution for removal of this component from EUI before we get to https://github.com/elastic/kibana/issues/107215.

Developers can consume this component by importing it:

```js
import { EuiCodeEditor } from '<path>/src/plugins/es_ui_shared/public';
```

I suggest creating a file called `shared_imports.ts` in the root of your plugin's `public` directory and re-exporting it there, for consumption by your plugin.

This PR also:
* Converts `src/plugins/es_ui_shared/public/components/json_editor/json_editor.tsx` to consume this transplanted `EuiCodeEditor`. `JsonEditor` is consumed by Index Management and Ingest Node Pipelines.
* Converts the Grok Debugger plugin to consume this component.

## Lazy-loading

This component is lazy-loaded to prevent it from bloating the EsUiShared plugin bundle. Here's how it appears when the network has been throttled to simulate a "Slow 3G" connection.

![lazy_load_code_editor](https://user-images.githubusercontent.com/1238659/129385178-c98e28c6-acd4-413f-b5d5-5f944f6cc01d.gif)

## Steps to test

This is a drop-in replacement for EuiCodeEditor since it's identical code. Testers can interact with Grok Debugger and the JSON editor in the Ingest Node Pipelines "Load from JSON" modal to verify that this change hasn't broken anything.
